src/peft/helpers.py

```diff
@@ -146,3 +148,56 @@ def check_if_peft_model(model_name_or_path: str) -> bool:
         is_peft_model = False
 
     return is_peft_model
+
+
+@contextmanager
+def set_adapter_scale(model, alpha):
+    """
+    Context manager to temporarily set the scaling of LoraLayer modules in a model.
+
+    This context manager iterates over the modules of the given model and scales the `scaling` attribute of `LoraLayer`
+    instances by the specified `alpha` factor. The original scaling values are restored when the context manager exits.
+
+    Args:
+        model: The model containing `LoraLayer` modules whose scaling is to be adjusted.
+        alpha (float): The scaling factor to be applied. Must be of type float.
+
+    Raises:
+        TypeError: If `alpha` is not of type float. ValueError: If the model does not contain any `LoraLayer`
```

Comment:
Let's add a line break for the two errors. Also, the `TypeError` doesn't really need documenting IMO.